filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
```

Comment:
Not sure I understand the examples above, wouldn't it be tagname1:foo*||bar*? For the && case it's not possible, you can't have tagname1:baz*&&w, value can't begin with baz and be equal to w for the same tag name. If you want begins with AND Ends with, you can do tagname1:baz*w. 

For the OR case, you can simply create 2 tag filters, one for tagname1:foo* and one for tagname1:bar* and put both filters as param to NewMultiFilter with LogicalOp as OR. It is cleaner, more flexible and gives us better control than trying to parse && and || inside the patterns. The patterns are globs right now and I'd prefer to keep it that way.